yaml.go

```diff
@@ -85,30 +85,33 @@ type Marshaler interface {
 //
 // See the documentation of Marshal for the format of tags and a list of
 // supported tag options.
-func Unmarshal(in []byte, out any) (err error) {
-	return unmarshal(in, out, false)
+func Unmarshal(in []byte, out any, opts ...Options) (err error) {
+	options := joinOptionsDefault(opts...)
+	return unmarshal(in, out, options)
 }
 
 // A Decoder reads and decodes YAML values from an input stream.
 type Decoder struct {
-	parser      *parser
-	knownFields bool
+	parser  *parser
+	options Options
 }
 
 // NewDecoder returns a new decoder that reads from r.
 //
 // The decoder introduces its own buffering and may read
 // data from r beyond the YAML values requested.
-func NewDecoder(r io.Reader) *Decoder {
+func NewDecoder(r io.Reader, opts ...Options) *Decoder {
+	options := joinOptionsDefault(opts...)
 	return &Decoder{
-		parser: newParserFromReader(r),
+		parser:  newParserFromReader(r),
+		options: options,
 	}
 }
 
 // KnownFields ensures that the keys in decoded mappings to
 // exist as fields in the struct being decoded into.
 func (dec *Decoder) KnownFields(enable bool) {
-	dec.knownFields = enable
+	dec.options.knownFields = &enable
```

Comment:
Here usually there should have a default configuration. With a loop on the variadic made of func(*config)